apps/ledger-live-desktop/src/newArch/features/ModularDrawer/utils/selectCurrency.ts

```diff
@@ -0,0 +1,32 @@
+import { Account, AccountLike } from "@ledgerhq/types-live";
+import { listAndFilterCurrencies } from "@ledgerhq/live-common/platform/helpers";
+
+import SelectCurrencyDrawer from "../components/SelectCurrencyDrawer";
+import { setDrawer } from "~/renderer/drawers/Provider";
+type SelectAccountAndCurrencyResult = {
+  account: AccountLike;
+  parentAccount?: Account;
+};
+export function selectCurrency(
```

Comment:
Why should this function be extracted from `useOpenAssetFlow` how will it be reused in a different context ? also why does it return a Promise ?